lib/src/main/java/io/ably/lib/push/Storage.java

```diff
@@ -0,0 +1,45 @@
+package io.ably.lib.push;
+
+import java.lang.reflect.Field;
+
+/**
+ * Interface for an entity that supplies key value store
+ */
+public interface Storage {
```

Comment:
I recognise that you're reflecting the Android APIs in your method names here, and in terms of using `get` and `put` as the verbs (instead of [my original](https://github.com/ably/ably-java/issues/593#issue-663543449) `read` and `write` suggestions) then I am happy with that.

What I don't think is needed, however, are the type suffixes as Java supports method overloads in this scenario and that leads to a cleaner API. i.e.:

- `putString` and `putInt` both rename to just `put` (or, maybe `putValue`)
- `getString` and `getInt` both rename to just `get` (or, maybe `getValue`)

Reply:
of course, great catch!

Reply:
https://github.com/ably/ably-java/pull/681/commits/16bf5233f352b5817f5afae1af1e95ab7e283c7f